oprf/group/group.go

```diff
@@ -0,0 +1,117 @@
+package group
+
+import (
+	"hash"
+	"math/big"
+)
+
+// CiphersuiteID is
+type CiphersuiteID int
+
+const (
+	// P256 is the group generated by P-256 elliptic curve.
+	P256 CiphersuiteID = iota
+)
+
+// Ciphersuite corresponds to the OPRF ciphersuite that is chosen. The
+
+// Ciphersuite corresponds to the OPRF ciphersuite that is chosen. The
+// Ciphersuite object determines the prime-order group (pog) that is used for
+// performing the OPRF operations, along with the different hash function
+// definitions.
+// Should be created using New, using a CiphersuiteID constant.
+type Ciphersuite struct {
+	// name of the ciphersuite
+	name string
+
+	// context
+	ctx string
+
+	// Suite instantiation for performing the OPRF operations.
+	suite Suite
+}
+
+// Suite is an interface that defines operations within additive
+// groups of prime order. This is the setting in which the OPRF operations
+// take place.
+type Suite interface {
+	// Creates a new Suite object
+	New(CiphersuiteID) (Suite, error)
+
+	// Returns the identifying name of the group
+	Name() string
+
+	// Returns the canonical (fixed) generator for defined group
+	Generator() *Point
+
+	// Returns the order of the canonical generator in the group.
+	Order() *big.Int
+
+	// Base hash function used in conjunction with the Ciphersuite
+	Hash() hash.Hash
+
+	// Returns k * G, where G is the canonical generator of the group, and k is
+	// some scalar value provided as input.
+	ScalarMult(*Scalar) (*Point, error)
```

Comment:
Why are these functions on the Suite interface? That seems a bit odd, no?

Reply:
Only this function or all of them? The generator, the order, the name, etc seem reasonable as they belong to the whole suite; but I agree that having the ScalarMult there is odd..